esmvaltool/recipes/examples/recipe_preprocessor_derive_test.yml

```
@@ -16,104 +16,232 @@ documentation:
+          - {dataset: CanESM2, ensemble: r1i1p1}
       rlns:
-        <<: *toz
-      clmtkisccp:
-        <<: *cloud
+        <<: *alb_cmip5
+      rlnstcs:
+        <<: *alb_cmip5
+      rlnst:
+        <<: *alb_cmip5
+      rlntcs:
+        <<: *alb_cmip5
+      rsns:
+        <<: *alb_cmip5
+      rsnstcsnorm:
+        <<: *alb_cmip5
+      rsnstcs:
+        <<: *alb_cmip5
+      rsnst:
+        <<: *alb_cmip5
+      rsntcs:
+        <<: *alb_cmip5
+      rsnt:
+        <<: *alb_cmip5
+      rtnt:
+        <<: *alb_cmip5
+      sispeed:
+        <<: *alb_cmip5
+        mip: day
+        start_year: 1979
+        end_year: 1979
+      sithick:
+        <<: *alb_cmip5
+        mip: day
+        start_year: 1979
+        end_year: 1979
+      sm:
+        <<: *alb_cmip5
+        mip: Lmon
+      swcre:
+        <<: *alb_cmip5
+      toz:
+        <<: *alb_cmip5
+      uajet:
+        <<: *alb_cmip5
+      vegfrac:
+        <<: *alb_cmip5
+        mip: Lmon
+        additional_datasets:
+          - {dataset: MPI-ESM-LR, ensemble: r1i1p1}
     scripts: null
 
-  diag6:
-    description: Test variable derivation
+  cmip6:
+    description: Test variable derivation for CMIP6 models.
     variables:
-      lwp:
-        <<: *toz
+      alb: &alb_cmip6
+        project: CMIP6
+        mip: Amon
+        exp: historical
+        start_year: 2004
+        end_year: 2005
+        derive: true
+        force_derivation: false
+        additional_datasets:
+          - {dataset: IPSL-CM6A-LR, ensemble: r1i1p1f1, grid: gr}
+      # Issues with required variables [#359]
+      # amoc:
+      #   <<: *alb_cmip6
+      #   mip: Omon
+      asr:
+        <<: *alb_cmip6
+      clhmtisccp: &cloud_cmip6
+        project: CMIP6
+        mip: CFmon
+        exp: hist-nat
+        start_year: 1987
+        end_year: 1988
+        derive: true
+        force_derivation: false
+        additional_datasets:
+          - {dataset: HadGEM3-GC31-LL, ensemble: r1i1p1f3, grid: gn}
+      clhtkisccp:
+        <<: *cloud_cmip6
       cllmtisccp:
-        <<: *cloud
-    scripts: null
-
-  diag7:
-    description: Test variable derivation
-    variables:
+        <<: *cloud_cmip6
       clltkisccp:
-        <<: *cloud
+        <<: *cloud_cmip6
+      clmmtisccp:
+        <<: *cloud_cmip6
+      clmtkisccp:
+        <<: *cloud_cmip6
+      co2s:
+        <<: *alb_cmip6
+        additional_datasets:
+          - {dataset: GFDL-ESM4, ensemble: r1i1p1f1, grid: gr1}
+      ctotal:
+        <<: *alb_cmip6
+        mip: Lmon
+      et:
+        <<: *alb_cmip6
+        mip: Lmon
+      lvp:
+        <<: *alb_cmip6
+      lwcre:
+        <<: *alb_cmip6
+      lwp:
+        <<: *alb_cmip6
       netcre:
-        <<: *toz
-    scripts: null
-
-  diag8:
-    description: Test sispeed
-    variables:
+        <<: *alb_cmip6
+      ohc:
+        <<: *alb_cmip6
+        mip: Omon
+        additional_datasets:
+          - {dataset: NorCPM1, ensemble: r2i1p1f1, grid: gr, institute: NCC}
```

vcello from NorCPM1 consists only of zeros which makes testing of ohc difficult. I would recommend to exchange this dataset with one from MPI-ESM1-2-LR:
```suggestion
          - {dataset: MPI-ESM1-2-LR, ensemble: r1i1p1f1, grid: gn, institute: MPI-M}
```

---

done!